codec-http2/src/test/java/io/netty/handler/codec/http2/Http2MultiplexCodecTransportTest.java

```diff
@@ -0,0 +1,141 @@
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.util.NetUtil;
+import io.netty.util.ReferenceCountUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.junit.Assert.assertFalse;
+
+public class Http2MultiplexCodecTransportTest {
+    private EventLoopGroup eventLoopGroup;
+    private Channel clientChannel;
+    private Channel serverChannel;
+    private Channel serverConnectedChannel;
+
+    @Before
+    public void setup() {
+        eventLoopGroup = new NioEventLoopGroup();
+    }
+
+    @After
+    public void teardown() {
+        if (clientChannel != null) {
+            clientChannel.close();
+        }
+        if (serverChannel != null) {
+            serverChannel.close();
+        }
+        if (serverConnectedChannel != null) {
+            serverConnectedChannel.close();
+        }
+        eventLoopGroup.shutdownGracefully(0, 0, MILLISECONDS);
+    }
+
+    @Test(timeout = 10000)
+    public void asyncSettingsAck() throws InterruptedException {
+        // The client expects 2 settings frames. One from the connection setup and one from this test.
+        final CountDownLatch serverAckLatch = new CountDownLatch(2);
+        final CountDownLatch clientSettingsLatch = new CountDownLatch(2);
+        final CountDownLatch serverConnectedChannelLatch = new CountDownLatch(1);
+        final AtomicReference<Channel> serverConnectedChannelRef = new AtomicReference<Channel>();
+        ServerBootstrap sb = new ServerBootstrap();
+        sb.group(eventLoopGroup);
+        sb.channel(NioServerSocketChannel.class);
+        sb.childHandler(new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) {
+                ch.pipeline().addLast(Http2MultiplexCodecBuilder.forServer(new HttpInboundHandler()).build());
+                ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                    @Override
+                    public void channelActive(ChannelHandlerContext ctx) {
+                        serverConnectedChannelRef.set(ctx.channel());
+                        serverConnectedChannelLatch.countDown();
+                    }
+
+                    @Override
+                    public void channelRead(ChannelHandlerContext ctx, Object msg) {
+                        if (msg instanceof Http2SettingsAckFrame) {
+                            serverAckLatch.countDown();
```

Comment:
nit: maybe also verify we do not receive more of these frame as expected.

Reply:
I changed the test to more carefully verify the first ACK and the second ACK, but I'll just leave it at that for now. Verifying more may require a timeout and this would be somewhat unreliable.